[Docs] Add manifest CMake integration and eval order details

ihnorton opened the pull request:
This PR adds a paragraph to the manifest docs, detailing the CMake integration and evaluation sequence.

ihnorton committed:
Add CMake integration and eval order details

Neumann-A commented:
Personally I think that this is missing a hint about the fact that there are other ways to include vcpkg into cmake. The important part is however that it needs to be integrated before the very first `find_x` call.
Furthermore it should be noted that changing the triplet requires a clean reconfigure since the results of `find_x` calls are stored in cache variables which won't get refreshed.

ihnorton committed:
Tweak wording and try to address PR feedback

ihnorton commented:
Hi @Neumann-A, thanks for the feedback.

> I think that this is missing a hint about the fact that there are other ways to include vcpkg into cmake.

I added a link to the [Using with cmake](https://github.com/microsoft/vcpkg#using-vcpkg-with-cmake) instructions in the main README. In the context of manifest mode, is there another approach than setting the toolchain file -- if so, could you give me a hint/pointer what you mean?

> Furthermore it should be noted that changing the triplet requires a clean reconfigure since the results of find_x calls are stored in cache variables which won't get refreshed.

Added, thanks. (as an side, one thing I am quite curious about is what other conditions require reconfigure or trigger rebuilds -- trying to understand what a local development workflow across multiple packages would look like, for example)

dg0yt commented on docs/users/manifests.md:
```diff
@@ -40,6 +41,10 @@ manifest mode.
 
+## How does it work?
+
+In manifest mode, vcpkg is integrated with CMake via a [toolchain](https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html), which must be [specified](https://github.com/microsoft/vcpkg#using-vcpkg-with-cmake) when the project is configured. The toolchain file is evaluated by CMake via [`include()`](https://cmake.org/cmake/help/latest/command/include.html) the first time CMake's [`proj ...
```
If this was broken into multiple lines, it might be easier to comment in GH.

dg0yt reviewed (commented):
The new paragraph comes with a general headline and describes CMake integration. Does this sufficiently reflect "How does it work?" AFAIU there is at least Visual Studio integration, and there might be others in the future.

I'm also wondering how this can be extended to cover the key difference between command line manifest mode and integrated manifest mode: Where is the installed tree?

BillyONeal commented:
I'm dropping reviewed from these because I'm not sure how they are affected by the ongoing transition to learn.microsoft.com. /cc @ras0219-msft @AugP

Cheney-W commented:
Close this PR and please track it with https://github.com/microsoft/vcpkg-docs/pull/33.

Cheney-W commented:
If you have any new documentation-related issues or prs, please submit them on https://github.com/microsoft/vcpkg-docs.
